### DISCUSSION TITLE
Global binaries

### DISCUSSION THREAD

**Discussion by merceyz** [yarnpkg/berry#3654] [Q&A]
I also met the same problem. 

```powershell
❯ node -v
v17.0.1
❯ npm -v
8.1.0
❯ yarn -v
3.1.0
❯ yarn add typescript
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed in 0s 437ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ typescript@npm:4.4.4 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ typescript@patch:typescript@npm%3A4.4.4#~builtin<compat/typescript>::version=4.4.4&hash=ddd1e8 can't be 
➤ YN0000: └ Completed in 0s 725ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: Done in 1s 254ms
❯ tsc -v
tsc: The term 'tsc' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
❯ yarn run tsc -v
Version 4.4.4
```

If I execute `tsc -v` directly in *Powershell Core*, it will throw out an error. But if I invoke `yarn run tsc -v`, it does response the version of *TypeScript* as I expected. **Note:** I didn't configure `tsc` for `scripts` in `package.json` (details stored [here](https://codechina.csdn.net/u011367208/typescript-notes/-/blob/ff7884b4f2a5d736cc2095e7ff9036d028dd894a/package.json)). 

I'm using *Plug'n'Play* and *Zero-Install* features from *Yarn v3.1.0*, is there something wrong with my configurations? Should I install *TypeScript* to global environment with `yarn dlx add typescript` or `yarn global add typescript` ?

_Originally posted by @Dragon1573 in https://github.com/yarnpkg/yarn/issues/648#issuecomment-951890746_

**Comment by liuliangsir** [comment]
@merceyz how to install yarn3 globally with npm

**Reply by Dragon1573** [reply]
According to the official "Getting Started" guide, it's NOT recommended to install Yarn v3.x globally.

Install Yarn v1.22.11 with `npm` and update to v3.x only inside a project is recommended.

If you REALLY need to install like that, I think you may try `npm install -g yarn@3.1.0`.

**Comment by theoparis** [comment]
> I also met the same problem.
> 
> ```powershell
> ❯ node -v
> v17.0.1
> ❯ npm -v
> 8.1.0
> ❯ yarn -v
> 3.1.0
> ❯ yarn add typescript
> ➤ YN0000: ┌ Resolution step
> ➤ YN0000: └ Completed in 0s 437ms
> ➤ YN0000: ┌ Fetch step
> ➤ YN0013: │ typescript@npm:4.4.4 can't be found in the cache and will be fetched from the remote registry
> ➤ YN0013: │ typescript@patch:typescript@npm%3A4.4.4#~builtin<compat/typescript>::version=4.4.4&hash=ddd1e8 can't be 
> ➤ YN0000: └ Completed in 0s 725ms
> ➤ YN0000: ┌ Link step
> ➤ YN0000: └ Completed
> ➤ YN0000: Done in 1s 254ms
> ❯ tsc -v
> tsc: The term 'tsc' is not recognized as a name of a cmdlet, function, script file, or executable program.
> Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
> ❯ yarn run tsc -v
> Version 4.4.4
> ```
> 
> If I execute `tsc -v` directly in _Powershell Core_, it will throw out an error. But if I invoke `yarn run tsc -v`, it does response the version of _TypeScript_ as I expected. **Note:** I didn't configure `tsc` for `scripts` in `package.json` (details stored [here](https://codechina.csdn.net/u011367208/typescript-notes/-/blob/ff7884b4f2a5d736cc2095e7ff9036d028dd894a/package.json)).
> 
> I'm using _Plug'n'Play_ and _Zero-Install_ features from _Yarn v3.1.0_, is there something wrong with my configurations? Should I install _TypeScript_ to global environment with `yarn dlx add typescript` or `yarn global add typescript` ?
> 
> _Originally posted by @Dragon1573 in [yarnpkg/yarn#648 (comment)](https://github.com/yarnpkg/yarn/issues/648#issuecomment-951890746)_

I wanted to use yarn over pnpm since a lot of projects prefer yarn, however this issue is preventing me from using yarn here as well. 
Since the global command has been removed in v2, I tried to use dlx and it does not work with different binary names. I'd like to see a flag to specify the binary name...
